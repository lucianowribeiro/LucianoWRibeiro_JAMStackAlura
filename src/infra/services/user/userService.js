import isStagingEnv from '../../env/isStagingEnv';
import HttpClient from '../../http/HttpClient';
import authService from '../auth/authService';
import loginService from '../login/loginService';

const BASE_URL = isStagingEnv
  ? 'https://instalura-api-git-master-omariosouto.vercel.app'
  : 'https://instalura-api-omariosouto.vercel.app';

const userService = {
  async getProfilePage(password, context) {
    try {
      const token = await authService(context).getToken();
      const session = await authService(context).getSession();
      const { user } = await loginService.login({ username: session.user.username, password });
      const response = await HttpClient(`${BASE_URL}/api/users/posts`, {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      return {
        user: {
          ...user,
        },
        posts: response.data,
      };
    } catch (error) {
      return null;
    }
  },
  getPassword() {
    return process.env.PASSWORD_DEFAULT;
  },
};
export default userService;
