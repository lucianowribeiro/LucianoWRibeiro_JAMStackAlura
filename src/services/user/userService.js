import isStagingEnv from '../../infra/env/isStagingEnv';
import HttpClient from '../../infra/http/HttpClient';
import authService from '../auth/authService';

const BASE_URL = isStagingEnv
  ? 'https://instalura-api-git-master-omariosouto.vercel.app'
  : 'https://instalura-api-omariosouto.vercel.app';

const userService = {
  async getProfilePage(context) {
    try {
      const token = await authService(context).getToken();
      console.log(token);
      const response = await HttpClient(`${BASE_URL}/api/users/posts`, {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      return {
        user: {
          likes: '100',
        },
        posts: response.data,
      };
    } catch (err) {
      throw new Error(err);
    }
  },
};
export default userService;
