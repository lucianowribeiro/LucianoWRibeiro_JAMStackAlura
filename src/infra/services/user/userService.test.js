import userService from './userService';

/* const dataFake = {
  _id: 'some id',
  photoUrl: 'https://images.endeavor.org.br/uploads/2015/05/simples-nacional1.jpg',
  description: 'some description',
  filter: 'some filter',
  user: 'some user',
  likes: [
    {
      _id: 'some id',
      user: 'some user',
    },
  ],
  createdAt: 'some data',
  updatedAt: 'some data 2',
  __v: 0,
};

async function HttpClientModule() {
  return {
    data: dataFake,
  };
} */

describe('postService', () => {
  describe('getProfilePage()', () => {
    describe('when user post an image', () => {
      describe('and succeed', () => {
        test('set the number of likes', async () => {
          const response = await userService.getProfilePage('senhasegura');
          await expect(response).toMatchSnapshot();
          await expect(response).toBeNull();
        });
      });
      describe('and it fails', () => {
        test('return null and keeping the same number of likes', async () => {
          const response = await userService.getProfilePage('');
          await expect(response).toMatchSnapshot();
          await expect(response).toBeNull();
        });
      });
    });
  });
});
