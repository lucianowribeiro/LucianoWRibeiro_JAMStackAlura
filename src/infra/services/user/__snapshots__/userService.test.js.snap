// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postService getProfilePage() when user post an image and it fails return null and keeping the same number of likes 1`] = `null`;

exports[`postService getProfilePage() when user post an image and succeed set the number of likes 1`] = `null`;
